src/test/java/com/google/sps/data/MapsRequestTest.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.data;
+
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.*;
+import com.google.sps.data.DeliverySlot;
+
+import java.io.IOException;
+import java.util.Date;
+import java.time.LocalTime;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests that responses from Google Maps API are parsed successfully.
+ */
+@RunWith(JUnit4.class)
+public class MapsRequestTest {
+  public DirectionsLeg getDirectionsLeg(int timeInSeconds) {
+    DirectionsLeg leg = new DirectionsLeg();
+    leg.duration = new Duration();
+    leg.duration.inSeconds = timeInSeconds;
+    return leg;
+  }
+
+  @Test
+  public void testDirectionsBetween2Points() throws BadRequestException, InterruptedException, DataNotFoundException, ApiException, IOException {
+    DirectionsResult result = new DirectionsResult();
+    result.routes = new DirectionsRoute[1];
+    result.routes[0] = new DirectionsRoute();
+    result.routes[0].legs = new DirectionsLeg[2];
+    result.routes[0].legs[0] = getDirectionsLeg(10);
+    result.routes[0].legs[1] = getDirectionsLeg(15);
+    Assert.assertEquals(25, MapsRequest.getDurationFromDirectionsResult(result));
```

Comment:
This is fine, but note you're not actually testing `getTimeInSecondsBetween2Points` (which is actually what you care about), but instead `getDurationFromDirectionsResult`.

Ideally you would test getTimeInSecondsBetween2Points by mocking DirectionsApi so that when calling `newRequest` you return an object which is mocked so that it always returns your harcoded result on `await()`. This way getDurationFromDirectionsResult can be private.

This might require using powermock's mockstatic to be able to mock DirectionsApi.

Reply:
Right, it makes sense, I will add that as a TODO.